specification/monitor/resource-manager/Microsoft.Insights/stable/2023-03-11/dataCollectionEndpoints_API.json

```diff
@@ -0,0 +1,781 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create a data collection endpoint": {
+            "$ref": "./examples/DataCollectionEndpointsCreate.json"
+          },
+          "Update a data collection endpoint": {
+            "$ref": "./examples/DataCollectionEndpointsUpdate.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "DataCollectionEndpoints"
+        ],
+        "summary": "Updates part of a data collection endpoint.",
+        "operationId": "DataCollectionEndpoints_Update",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DataCollectionEndpointNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "The payload",
+            "schema": {
+              "$ref": "#/definitions/ResourceForUpdate"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Data collection endpoint was successfully updated",
+            "schema": {
+              "$ref": "#/definitions/DataCollectionEndpointResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "./dataCollectionRules_API.json#/definitions/ErrorResponseCommonV2"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Update data collection endpoint": {
+            "$ref": "./examples/DataCollectionEndpointsPatch.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "DataCollectionEndpoints"
+        ],
+        "summary": "Deletes a data collection endpoint.",
+        "operationId": "DataCollectionEndpoints_Delete",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DataCollectionEndpointNameParameter"
+          },
+          {
+            "in": "query",
+            "name": "deleteAssociations",
```

Comment:
@ms-henglu is this a hard requirement from ARM? If not, can we skip it because I don't see any other better way to pass this argument (if not in the query)